tests/test_dir_traversal.py

```diff
@@ -257,6 +267,484 @@ class DirectoryTraversalVector(Generic[T]):
             ],
         ],
     ),
+    DirectoryTraversalVector(
+        "ignored_dot_dirs__are_not_traversed",
+        given=[
+            File(".venv/sub/file"),
+            File("dir/.venv/sub/file"),
+            File("dir/foo.py"),
+        ],
+        ignore_patterns=[".*"],  # ignore all paths that start with "."
+        expect=[
+            ExpectedTraverseStep(".", subdirs=["dir"]),  # ignored_subdirs=[".venv"]
+            ExpectedTraverseStep("dir", files=["foo.py"]),  # ignored_subdirs=[".venv"]
+        ],
+    ),
+    #
+    # Testing ignore patterns
+    #
+    # The following tests are based on the pattern format rules listed in
+    # `git help ignore` (https://git-scm.com/docs/gitignore#_pattern_format).
+    #
+    # A blank line matches no files, so it can serve as a separator for
+    # readability.
+    DirectoryTraversalVector(
+        "gitignore_parsing__disregard_blank_lines",
+        given=[File("dir/file")],
+        ignore_patterns=["", ""],
+        ignore_exceptions=[IgnoreRuleMissing, IgnoreRuleMissing],
```

Comment:
We're checking that we get `IgnoreRuleMissing` raised twice, once for each empty pattern above.

[In the test below](https://github.com/tweag/FawltyDeps/pull/388/files/142b91a7559c389414ac807556daeff5d3919074#diff-3aa986ddb8b14d40d3559e3b54826bfc5b43b56d33d849bd21f82378c8723764R763) we pass each ignore pattern separately to the parser and recording each exception raised. I wanted to make sure that the first exception is not messing up some internal state that would cause problems for later ignore patterns. (When we get around to parsing gitignore files, we will encounter valid ignore patterns interspersed with empty lines, and we need to skip the empty lines and keep parsing.)